File: SEGUNDO/C2/PDOO/backup/Deep_space/src/deepspace/Loot.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */
public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    Loot(int supplies, int weapons,int shields, int hangars, int medals){
    this.nSupplies=supplies;
    this.nWeapons=weapons;
    this.nShields=shields;
    this.nHangars=hangars;
    this.nMedals=medals;
}
    
    public int getNSupplies(){
        return this.nSupplies;
    }
    
    public int getNWeapons(){
        return this.nWeapons;
    }
    
    public int getNShields(){
        return this.nShields;
    }
    
    public int getNHangars(){
        return this.nHangars;
    }
    
    public int getNMedals(){
        return this.nMedals;
    }
    
    LootToUI getUIversion(){
        return new LootToUI(this);
    }
    
    @Override
    public String toString(){
        String salida = "Suplementos: " + nSupplies + "\nNº de Armas: " + nWeapons + "\nNº Escudos: " + nShields + "\nNº Hangares: " + nHangars;
        salida = salida + "\nNº de Medallas: " + nMedals + "\n";
        
        return salida;        
    }
    
}

File: SEGUNDO/C1/NetBeansProjects/DeepSpace_ruby/pkg/DeepSpace_ruby-0.0.1/lib/loot.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'LootToUI.rb'
module Deepspace

class Loot
  def initialize(nSupplies,nWeapons,nShields,nHangars,nMedals)
    @nSupplies=nSupplies;
    @nWeapons=nWeapons;
    @nShields=nShields;
    @nHangars=nHangars;
    @nMedals=nMedals; 
  end
  
  attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals
  
  def to_s
    "Numero de Suplementos: #{@nSupplies}\nNumero de Armas: #{@nWeapons}\nNumero de escudos: #{@nShields}\nNumero de Hangares: #{@nHangars}\nNumero de Medallas: #{@nMedals}"
  end
  
  def getUIVersion
    return LootToUI.new(self)
  end
  
end

end
